src/forms/medication-dispense-review.component.tsx

```diff
@@ -164,36 +135,27 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
   );
 
   // check to see if the current dispense would be a substitution, and update accordingly
-  useEffect(() => {
-    if (
-      medicationRequest?.medicationReference?.reference &&
-      medicationDispense?.medicationReference?.reference &&
-      medicationRequest.medicationReference.reference != medicationDispense.medicationReference.reference
-    ) {
-      setIsSubstitution(true);
-      updateMedicationDispense({
-        ...medicationDispense,
-        substitution: {
-          ...medicationDispense.substitution,
-          wasSubstituted: true,
-        },
-      });
+  const isSubstitution =
+    medicationRequest?.medicationReference?.reference &&
+    medicationDispense?.medicationReference?.reference &&
+    medicationRequest.medicationReference.reference != medicationDispense.medicationReference.reference;
```

Comment:
```suggestion
  const isSubstitution = useMemo(
    () =>
      medicationRequest?.medicationReference?.reference &&
      medicationDispense?.medicationReference?.reference &&
      medicationRequest.medicationReference.reference != medicationDispense.medicationReference.reference,
    [
      medicationRequest?.medicationReference?.reference,
      medicationDispense?.medicationReference?.reference,
    ],
  );
  
```

nit: Minor thing, but we could consider memoizing `isSubstitution` so it's not recalculated on every render. That said, it's not the most complex computation, hence the nit prefix.